extensions/ql-vscode/src/compare/resultsDiff.ts

```diff
@@ -0,0 +1,58 @@
+import { RawResultSet } from '../adapt';
+import { QueryCompareResult } from '../interface-types';
+
+/**
+ * Compare the rows of two queries. Use deep equality to determine if
+ * rows have been added or removed across two invocations of a query.
+ *
+ * Assumptions:
+ *
+ * 1. Queries have the same sort order
+ * 2. Queries have same number and order of columns
+ * 3. Rows are not changed or re-ordered, they are only added or removed
+ *
+ * @param fromResults the source query
+ * @param toResults the target query
+ *
+ * @throws Error when:
+ *  1. number of columns do not match
+ *  2. If either query is empty
+ *  3. If the queries are 100% disjoint
```

Comment:
When I do a comparison of disjoint queries, the comparison view is forced visible, and I do get an error message saying the results are disjoint, but the comparison view still contains whichever comparison it did before, which I think is slightly misleading. I think I'd prefer --- but would be happy deferring to a follow-up PR if this is at all complicated --- having one of the possible states of the comparison view actually being "results disjoint" as opposed to `window.showErrorMessage`.

Reply:
Hmm...right. I should fix that.

Reply:
There are a few other error states. What I will do is ensure that the compare view will hide the columns and display the error message in this case.

Reply:
Yeah, that sounds great.